src/sentry/grouping/ingest.py

```diff
@@ -256,6 +256,23 @@ def run_primary_grouping(
                 job["event"].data.data, project
             )
 
+            # TODO: For new (non-reprocessed) events, we read the grouping config off the event
+            # rather than from the project. But that grouping config is put there by Relay after
+            # looking it up on the project. Are these ever not the same? If we don't ever see this
```

Comment:
I think these could potentially be different right after the grouping config is changed for a project and relay still has the old project config cached

Reply:
That's a good point. That said, that's even more reason to just always pull from the project. If your scenario happened, the data on the event would be out of date.